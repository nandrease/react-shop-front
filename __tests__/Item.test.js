import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
	id: '49EO',
	title: 'Suzuki GSXR',
	price: 2000,
	description: '750cc hästi sõitev ratas',
	image: 'bike.jpg',
	largeImage: 'bike_large.jpg'
};

describe('<Item/>', () => {
	it('renders and matches the snapshot', () => {
		const wrapper = shallow(<ItemComponent item={fakeItem} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
	// it('renders and displays properly', () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />);
	// 	const PriceTag = wrapper.find('PriceTag');
	// 	const Title = wrapper.find('Title a');
	// 	const Img = wrapper.find('img');
	// 	expect(PriceTag.children().text()).toBe('20 €');
	// 	expect(Title.text()).toBe(fakeItem.title);
	// 	expect(Img.props().src).toBe(fakeItem.image);
	// 	expect(Img.props().alt).toBe(fakeItem.title);
	// });

	// it('renders out the buttons properly', () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />);
	// 	const buttonList = wrapper.find('.buttonList');
	// 	// console.log(buttonList.debug());
	// 	expect(buttonList.children()).toHaveLength(3);
	// 	expect(buttonList.find('Link')).toHaveLength(1);
	// 	expect(buttonList.find('AddToCart')).toBeTruthy();
	// 	expect(buttonList.find('DeleteItem').exists()).toBe(true);
	// });
});
