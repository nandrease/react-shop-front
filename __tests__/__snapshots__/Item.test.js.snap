// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Suzuki GSXR"
    src="bike.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "49EO",
          },
        }
      }
    >
      <a>
        Suzuki GSXR
      </a>
    </Link>
  </Title>
  <PriceTag>
    20Â â‚¬
  </PriceTag>
  <p>
    750cc hÃ¤sti sÃµitev ratas
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "49EO",
          },
        }
      }
    >
      <a>
        Edit ðŸ¤£ðŸ§¨
      </a>
    </Link>
    <AddToCart
      id="49EO"
    />
    <DeleteItem
      id="49EO"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
